lib/install.mjs

```diff
@@ -0,0 +1,132 @@
+import { createWriteStream, existsSync, rmSync, renameSync } from 'node:fs';
+import { get } from 'node:https';
+import { dirname, resolve } from 'node:path';
+import { arch, env, platform, exit } from 'node:process';
+import { fileURLToPath, URL } from 'node:url';
+
+import compressing from 'compressing';
+import progress from 'cli-progress';
+import semver from 'semver';
+
+import nodeManifest from "../package.json" assert { type: "json" };
+
+/**
+ * NW.js build flavor
+ * 
+ * @type {'sdk' | 'normal'}
+ */
+let buildType = env.npm_config_nwjs_build_type || env.NWJS_BUILD_TYPE || 'normal';
+
+// Parse Node manifest version.
+let parsedVersion = semver.parse(nodeManifest.version);
+let versionString = `${parsedVersion.major}.${parsedVersion.minor}.${parsedVersion.patch}`;
+
+// Check if version is a prelease.
+if (parsedVersion.prerelease !== undefined && typeof parsedVersion.prerelease[0] === 'string') {
+  let prerelease = parsedVersion.prerelease[0].split('-');
+  if (prerelease.length > 1) {
+    prerelease = prerelease.slice(0, -1);
+  }
+  versionString = `${versionString}-${prerelease}`;
```

Comment:
The original logic for this line would be equivalent to:
```suggestion
  versionString = versionString + '-' + prerelease.join('-');
```

Not sure if the extra `.join('-')` is actually needed, just pointing out in case it is an oversight.

Reply:
I think this should be `versionString = [ versionString, prerelease ].join('-')`